src/parseProxyOps.ts

```diff
@@ -1,21 +1,32 @@
 import { subscribe } from 'valtio/vanilla';
 
 type Op = Parameters<Parameters<typeof subscribe>[1]>[0][number];
-type ArrayOp = [
+export type ArrayOp = [
   op: 'insert' | 'set' | 'delete',
   index: number,
   value1: unknown,
   value2: unknown,
 ];
-
-export const parseProxyOps = (ops: Op[]): ArrayOp[] => {
-  const arrayOps: ArrayOp[] = ops.flatMap((op) => {
-    if (op[0] === 'resolve' || op[0] === 'reject') return [];
-    const index = Number(op[1][op[1].length - 1]);
-    if (!Number.isFinite(index)) return [];
-    return [[op[0], index, op[2], op[3]]];
+export const parseProxyOps = (ops: Op[]): Record<string, ArrayOp[]> => {
```

Comment:
Isn't there a way to do it without changing the signature?

Reply:
I could maybe call it `partitionedProxyOps`, call that from `subscribeP` and leave the original as is, or you were thinking of a different option?

Reply:
I'm thinking not changing the exported function at all. I might be missing your refactoring point though.

Reply:
> The refactor makes sure parseProxyOps is called one time. And arrayOps are processed on a per path basis.

Maybe my question is whether we can ensure calling the function once, without processing on a per-path basis.

Reply:
Got you. Yes, we could do that. But then there is no escaping the redundant runs that happen in the `aOps` forEach.
So below I am just printing the number of iterations for the below example
```js
import * as Y from 'yjs';
import { proxy } from 'valtio/vanilla';
import { bind } from '../src/old/index';

test.only('dev', async () => {
  const doc = new Y.Doc();
  const a = doc.getMap('arr');

  const p = proxy<any>({
    x: [1, 2, 3, 4, 5, 6, 7, 8, 9],
    y: [1, 2, 3, 4, 5, 6, 7, 8, 9],
    z: [1, 2, 3, 4, 5, 6, 7, 8, 9],
    i: [1, 2, 3, 4, 5, 6, 7, 8, 9],
    j: [1, 2, 3, 4, 5, 6, 7, 8, 9],
    k: [1, 2, 3, 4, 5, 6, 7, 8, 9],
  });
  bind(p, a);
  (() => {
    p.x.reverse();
    p.y.reverse();
    p.z.reverse();
    p.i.reverse();
    p.j.reverse();
    p.k.reverse();
  })();

  expect(true).toBe(true);
});

```
Before this refactor
![image](https://github.com/dai-shi/valtio-yjs/assets/9935834/a8e5b608-f8d0-47d4-8b06-7567da29f26e)
after
![image](https://github.com/dai-shi/valtio-yjs/assets/9935834/fa1b3194-8063-4dfe-9a29-6236490486e9)

Reply:
I would use a WeakMap for `snapshot(proxyObj)` to avoid the duplication.

I wonder if we can write a failing test firstly.